src/Sulu/Bundle/MediaBundle/Controller/MediaController.php

```diff
@@ -96,10 +104,18 @@ function ($id) use ($locale, $mediaManager) {
     public function cgetAction(Request $request)
     {
         try {
+            $collectionId = $request->get('collection');
+
+            if ($collectionId) {
+                $collection = $this->getCollectionManager()->getById($collectionId, $this->getLocale($request));
```

Comment:
I don't know a different way than loading the collection here, in order to be able to check if it is a system collection, and check if the user is allowed to see the content... /cc @chirimoya